plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt

```diff
@@ -187,6 +186,22 @@ class LiquidDataflowTests {
     )
   }
 
+  @Test
+  fun `when with val`() {
+    """
+      ${imports()}
+      fun bar(x: Int): Int {
+        pre( x >= 0 ) { "x is >= 0" }
+        return (when (val y = x + 1) {
```

Comment:
Should we test also `is` and the case where multiple `is` expressions appear in the same branch?